lib/hackmode/realmode.js

```diff
@@ -0,0 +1,1159 @@
+            const metaId = this._metaIdFromTuple(data.entryAddress, data.attribute, metaContent)
+            // Store hash of metaContent in DHT
+            const contentHash = getHash(metaId)
+
+            dht.post(DhtEvent.dataHoldRequest(data.entryAddress, contentHash))
+
+            // Store in Core if Core isn't already holding it
+            if (this._hasMeta(dnaAddress, metaId)) {
+              log.t('metaContent is known:', metaContent)
+              continue
+            }
+            log.t('metaContent is unknown:', metaContent)
+            this._bookkeepAddress(this._storedMetaBook, dnaAddress, metaId)
+            toStoreList.push(metaContent)
+          }
+          // Send toStoreList to Core
+          // log.t(this.me() + ' Sending IPC handleStoreMeta: ', toStoreList)
+          this._ipcSend('json', {
+            method: 'handleStoreMeta',
+            dnaAddress,
+            providerAgentId: data.providerAgentId,
+            entryAddress: data.entryAddress,
+            attribute: data.attribute,
+            contentList: toStoreList
+          })
+        }
+      })
+      this._memory[dnaAddress] = mem
+    }
+    return this._memory[dnaAddress]
+  }
+
+  /**
+   *  Check if agent is tracking dna.
+   *  If not, will try to send a FailureResult back to sender (if sender info is provided).
+   *  Returns transportId of receiverAgentId if agent is tracking dna.
+   *  @private
+   */
+  _getTransportIdOrFail (dnaAddress, receiverAgentId, senderAgentId, requestId) {
+    const peer = this._getDhtRef(dnaAddress).getPeerLocal(receiverAgentId)
+    if (!peer) {
+      // Send FailureResult back to IPC, should be senderAgentId
+      log.e(this.me() + ' #### CHECK FAILED for (agent) "' + receiverAgentId + '" + (DNA) "' + dnaAddress + '" (sender: ' + senderAgentId + ')')
+      this._ipcSend('json', {
+        method: 'failureResult',
+        dnaAddress: dnaAddress,
+        _id: requestId,
+        toAgentId: senderAgentId,
+        errorInfo: 'No routing for agent id "' + receiverAgentId + '"'
+      })
+      return null
+    }
+    const peerId = peer.peerTransport.substring(5) // remove 'hc://'
+    log.t(this.me() + ' oooo CHECK OK for (agent)"' + receiverAgentId + '" + (DNA) "' + dnaAddress + '" = ' + this.nick(peerId))
+    return peerId
+  }
+
+  /**
+   * @private
+   */
+  _untrack (dnaAddress, agentId) {
+    log.t(this.me() + ' _untrack() for "' + agentId + '" for DNA "' + dnaAddress + '"')
+    this._ipcRemoveTrack(agentId, dnaAddress)
+  }
+
+  /**
+   * @private
+   */
+  async _track (dnaAddress, agentId) {
+    log.t(this.me() + ' _track', dnaAddress, agentId)
+    // Bookkeep agentId -> dnaAddress
+    this._ipcAddTrack(agentId, dnaAddress)
+    // Init DHT for this DNA and bookkeep agentId -> PeerId for self
+    if (!(dnaAddress in this._dhtPerDna)) {
+      await this._initDht(dnaAddress, agentId)
+    } else {
+      const peerHoldEvent = DhtEvent.peerHoldRequest(
```

Comment:
oh, tricky. This might make sense for now. I was thinking we'd have a different dht for each dna/agentId combo... but certainly with the full-sync that isn't needed. We may have to revisit this with rrdht given each node should be storing data near their neighborhood, but with this we'll only be storing it in the _thisPeer`s neighborhood.

Reply:
yep. But will really have several different agentIds for one n3h instance?
Does the conductor handle several agentIds? If we have one n3h per conductor, I don't see how we can get several agents per n3h.

Reply:
you can currently configure as many agents with any dna you want in a single conductor config... so you can totally have 100s of agents with the same dna... not sure why anyone would do that though...